New: Blog post announcing rules policies changes (#732)
eslint/website: _posts/2020-05-22-changes-to-rules-policies.md

---
layout: post
title: Changes to our rules policies
tags:
  - rules
  - policies
---
# Changes to our Rules Policies

> When ESLint was first created, the idea was to have a small set of 20-30 core rules that could get everyone started, and then to let the community build out its own rules to fill in any remaining gaps. We did, however, continue implementing core rules as it seemed helpful to the point where today there are 280 core rules. At any given time, more than half of open issues and pull requests are related to rules, and unfortunately, the maintenance overhead is not sustainable any longer. As such, we're making some changes to how we evaluate rule-related requests going forward.

Reviewer: Right now this could be read as us saying "not our problem anymore", which I could see causing some blowback. What do you think of more explicitly framing this as "we're now able to rely on community rules, so the core team can focus on core priorities"? Perhaps offer a carrot like parallel linting. Possible wording to elaborate on that idea:

```suggestion
When ESLint was first created, the idea was to have a small set of 20-30 core rules that could get everyone started, and then to let the community build out its own rules to fill in any remaining gaps. We did, however, continue implementing core rules as it seemed helpful to the point where today there are 280 core rules. At any given time, more than half of open issues and pull requests are related to rules that take up a significant portion of the team's time.

ESLint now benefits from a thriving ecosystem of rule-focused plugins like `eslint-plugin-node`, `eslint-plugin-import`, `eslint-plugin-react`, and many others. Tools like astexplorer.net and numerous tutorials have made it easier than ever for developers to write their own rules. The core team no longer needs to build up a critical mass of basic rules, so we can prioritize our time on long-awaited core features like parallel linting that will benefit the whole community. We're making some changes to how we prioritize changes to built-in rules, but as always, if a rule isn't quite right for your situation, you're encouraged to modify it as a custom rule for your own project and share it with the rest of the community. 
```

* **New rules are limited** - we will only accept new rules when they relate to new ECMAScript features that reached stage 4 in the preceding 12 months. The community relies on ESLint to help teach them the correct way to use new language features, and we want to continue doing that. We won't be accepting new rules that are suggestions or preferences unrelated to new ECMAScript features.
* **No new rules that only disallow syntax** - we already have [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax), which should work for most cases. Otherwise, people can create their own rules. We do have some legacy rules that just disallow syntax (like [`no-undefined`](https://eslint.org/docs/rules/no-undefined)) that we will keep, but we won't be adding any more.
* **Stylistic rules are frozen** - we won't be adding any more options to stylistic rules. We've learned that there's no way to satisfy everyone's personal preferences, and most of the rules already have a lot of difficult-to-understand options. Stylistic rules are those related to spacing, conventions, and generally anything that does not highlight an error or a better way to do something.

> * **New rule options must be implemented by community members** - People can still propose new options for existing core non-stylistic rules, and we will still evaluate them as usual. However, these approved options will need to be implemented by the community and won't be part of the official ESLint roadmap. 

Author: Fine by me.

---
layout: post
title: Changes to our rules policies
tags:
  - rules
  - policies
---
# Changes to our Rules Policies

When ESLint was first created, the idea was to have a small set of 20-30 core rules that could get everyone started, and then to let the community build out its own rules to fill in any remaining gaps. We did, however, continue implementing core rules as it seemed helpful to the point where today there are 280 core rules. At any given time, more than half of open issues and pull requests are related to rules that take up a significant portion of the team's time.

> ESLint now benefits from a thriving ecosystem of rule-focused plugins like `eslint-plugin-node`, `eslint-plugin-import`, `eslint-plugin-react`, and many others. Tools like astexplorer.net and numerous tutorials have made it easier than ever for developers to write their own rules. The core team no longer needs to build up a critical mass of basic rules, so we can prioritize our time on long-awaited core features like parallel linting that will benefit the whole community. We're making some changes to how we prioritize changes to built-in rules, but as always, if a rule isn't quite right for your situation, you're encouraged to modify it as a custom rule for your own project and share it with the rest of the community. 

Author: ```suggestion
ESLint now benefits from a thriving ecosystem of rule-focused plugins like `eslint-plugin-node`, `eslint-plugin-import`, `eslint-plugin-react`, and many others. Tools like [astexplorer.net](https://astexplorer.net)and numerous tutorials have made it easier than ever for developers to write their own rules. The core team no longer needs to build up a critical mass of basic rules, so we can prioritize our time on long-awaited core features like parallel linting that will benefit the whole community. We're making some changes to how we prioritize changes to built-in rules, but as always, if a rule isn't quite right for your situation, you're encouraged to modify it as a custom rule for your own project and share it with the rest of the community. 
```

Reviewer: Missing a space between the link and the following "and"

> ESLint now benefits from a thriving ecosystem of rule-focused plugins like `eslint-plugin-node`, `eslint-plugin-import`, `eslint-plugin-react`, and many others. Tools like [astexplorer.net](https://astexplorer.net)and numerous tutorials have made it easier than ever for developers to write their own rules. The core team no longer needs to build up a critical mass of basic rules, so we can prioritize our time on long-awaited core features like parallel linting that will benefit the whole community. We're making some changes to how we prioritize changes to built-in rules, but as always, if a rule isn't quite right for your situation, you're encouraged to modify it as a custom rule for your own project and share it with the rest of the community. 

Author: ```suggestion
ESLint now benefits from a thriving ecosystem of rule-focused plugins like `eslint-plugin-node`, `eslint-plugin-import`, `eslint-plugin-react`, and many others. Tools like [astexplorer.net](https://astexplorer.net) and numerous tutorials have made it easier than ever for developers to write their own rules. The core team no longer needs to build up a critical mass of basic rules, so we can prioritize our time on long-awaited core features like parallel linting that will benefit the whole community. We're making some changes to how we prioritize changes to built-in rules, but as always, if a rule isn't quite right for your situation, you're encouraged to modify it as a custom rule for your own project and share it with the rest of the community. 
```
